lib/project-view.coffee

```diff
@@ -124,8 +138,10 @@ module.exports = ProjectView =
     # Shorten root path if possible
     userHome = fs.getHomeDirectory()
     normRootPath = path.normalize(rootPath)
-    if normRootPath.indexOf(userHome) is 0
-      # Use also tilde in case of Windows as synonym for the home folder
+    if atom.config.get 'project-view.regexMatch'
```

Comment:
To make the logic more fail save better check if `@regex` is null. I.e. set only `@regex` if `project-view.regexMatch` isn't an empty string. Otherwise `@regex` should be always null.
Then you could do
```
if @regex?
  normRootPath.replace(@regex),
    atom.config.get('project-view.rewrite'))
```